spec.html

```diff
@@ -23991,16 +24144,273 @@ <h1>Runtime Semantics: Evaluation</h1>
+    </emu-grammar>
+
+    <emu-clause id="sec-decoratorevaluation" type="sdo">
+      <h1>Runtime Semantics: DecoratorEvaluation ( ): either a normal completion containing an ECMAScript language value or an abrupt completion</h1>
+      <dl class="header">
+      </dl>
+      <emu-grammar>Decorator : `@` DecoratorMemberExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |MemberExpression| that is covered by |DecoratorMemberExpression|.
+        1. Let _ref_ be ? Evaluation of _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+      <emu-grammar>Decorator : `@` DecoratorCallExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |CallMemberExpression| that is covered by |DecoratorCallExpression|.
+        1. Let _ref_ be ? Evaluation of _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+      <emu-grammar>Decorator : `@` DecoratorParenthesizedExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |MemberExpression| that is covered by |DecoratorParenthesizedExpression|.
+        1. Let _ref_ be ? Evaluation of _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-decoratorlistevaluation" type="sdo">
+      <h1>Runtime Semantics: DecoratorListEvaluation ( ): either a normal completion containing a List of ECMAScript language values or an abrupt completion</h1>
+      <dl class="header">
+      </dl>
+      <emu-grammar>DecoratorList : DecoratorList? Decorator</emu-grammar>
+      <emu-alg>
+        1. If |DecoratorList| is present, then
+          1. Let _leftValue_ be ? DecoratorListEvaluation of |DecoratorList|.
+        1. Else,
+          1. Let _leftValue_ be a new empty List.
+        1. Let _rightValue_ be ? DecoratorEvaluation of |Decorator|.
+        1. Prepend _rightValue_ to _leftValue_.
+        1. Return _leftValue_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratoraccessobject" type="abstract operation">
+      <h1>
+        CreateDecoratorAccessObject (
+          _kind_: ~field~, ~method~, ~accessor~, ~getter~, or ~setter~,
+          _name_: either a String, a Symbol, or a Private Name,
+        ): an Object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Returns an object with appropriate accessor functions for the value, for use in a decorator context object.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _accessObj_ be OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is ~field~, ~method~, ~accessor~, or ~getter~, then
+          1. Let _getterClosure_ be a new Abstract Closure with parameters (_obj_) that captures _name_ and performs the following steps when called:
+            1. If _obj_ is not an Object, throw a *TypeError* exception.
+            1. If _name_ is either a String or a Symbol, return ? Get(_obj_, _name_).
+            1. Else, return ? PrivateGet(_name_, _obj_).
```

Comment:
i just noticed this, but it seems really weird to have PrivateGet/Set take name first instead of object first (like Get/Set do).

Reply:
PrivateGet/Set *do* take the object first. This invocation has the args in the wrong order.